web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackHelper.php

```diff
@@ -0,0 +1,227 @@
+use Drupal\core_event_dispatcher\Event\Entity\EntityInsertEvent;
+use Drupal\core_event_dispatcher\Event\Entity\EntityUpdateEvent;
+use Drupal\webform\WebformSubmissionInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Project track helper.
+ */
+final class ProjectTrackHelper extends AbstractHelper implements EventSubscriberInterface {
+
+  /**
+   * The project track storage.
+   *
+   * @var \Drupal\ai_screening_project_track\ProjectTrackStorageInterface
+   */
+  private readonly ProjectTrackStorageInterface $projectTrackStorage;
+
+  public function __construct(
+    private readonly TimeInterface $time,
+    private readonly EntityTypeManagerInterface $entityTypeManager,
+    LoggerChannel $logger,
+  ) {
+    parent::__construct($logger);
+    $this->projectTrackStorage = $this->entityTypeManager->getStorage('project_track');
+  }
+
+  /**
+   * Get data from track.
+   *
+   * @param \Drupal\ai_screening_project_track\ProjectTrackInterface $track
+   *   *   The track.
+   * @param string $key
+   *   The key.
+   *
+   * @return mixed
+   *   The data or null.
+   *
+   *   Use self::hasTrackData() to check if data is actually set (and possibly
+   *   null).
+   *
+   * @see self::hasTrackData()
+   */
+  public function getToolData(ProjectTrackInterface $track, ?string $key = NULL): mixed {
+    try {
+      $data = $track->getToolData();
+
+      return NULL === $key ? $data : ($data[$key] ?? NULL);
+    }
+    catch (\Exception $exception) {
+      $this->logException($exception, __METHOD__, [
+        'track' => $track,
+        'key' => $key,
+      ]);
+    }
+
+    return NULL;
+  }
+
+  /**
+   * Check if track has data.
+   *
+   * @param \Drupal\ai_screening_project_track\ProjectTrackInterface $track
+   *   The track.
+   * @param string $key
+   *   The key.
+   *
+   * @return bool
+   *   True if data for key exists.
+   */
+  public function hasTrackData(ProjectTrackInterface $track, string $key): bool {
+    try {
+      $data = $track->getToolData();
+
+      return array_key_exists($key, $data);
+    }
+    catch (\Exception $exception) {
+      $this->logException($exception, __METHOD__, [
+        'track' => $track,
+        'key' => $key,
+      ]);
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Add data to track.
+   *
+   * @param \Drupal\ai_screening_project_track\ProjectTrackInterface $track
+   *   The track.
+   * @param string $key
+   *   The key.
+   * @param mixed $value
+   *   The value.
+   */
+  public function setTrackData(ProjectTrackInterface $track, string $key, mixed $value): void {
+    try {
+      $track->setToolData([$key => $value] + $track->getToolData());
+    }
+    catch (\Exception $exception) {
+      $this->logException($exception, __METHOD__, [
+        'track' => $track,
+        'key' => $key,
+      ]);
+    }
+  }
+
+  /**
+   * Add a webform submission to a track.
+   */
+  public function addSubmission(
```

Comment:
Why no params description in this?

Reply:
They don't add any value … But now I've added them.